application/core/LSYii_Application.php

```diff
@@ -591,4 +591,173 @@ private function setSessionByDB($aApplicationConfig)
             'lifetime' => $lifetime
         ]);
     }
+
+    /**
+     * Creates a table based on another
+     *
+     * @param string $table
+     * @param string $pattern
+     * @param array $columns
+     * @param array $where
+     * @return integer number of rows affected by the execution.
+     * @throws CDbException execution failed
+     */
+    public function createTableFromPattern($table, $pattern, $columns = [], $where = [])
+    {
+        if (!is_array($columns)) {
+            $columns = [];
+        }
+        if (!is_array($where)) {
+            $where = [];
+        }
+        $whereClause = "";
+        $criterias = [];
+        if (count($where)) {
+            foreach ($where as $field => $value) {
+                $criterias[] = $this->db->quoteColumnName($field) . " = " . $this->db->quoteValue($value);
+            }
+            $whereClause = " WHERE " . implode(" AND ", $criterias);
+        }
+        if (count($columns)) {
+            foreach ($columns as $index => $column) {
+                if (!ctype_alnum($column)) {
+                    $columns[$index] = $this->db->quoteColumnName($column);
+                }
+            }
+            $command = "CREATE TABLE " . $this->db->quoteTableName($table) . " AS SELECT " . implode(",", $columns) . " FROM " . $this->db->quoteTableName($pattern) . $whereClause;
+        } else {
+            $command = "CREATE TABLE " . $this->db->quoteTableName($table) . " LIKE " . $this->db->quoteTableName($pattern) . ";";
+        }
+        return $this->db->createCommand($command)->execute();
+    }
+
+    /**
+     * Gets the unchanged columns
+     *
+     * @param int $sid
+     * @param int $qTimestamp
+     * @param int $sTimestamp
+     * @return array all rows of the query result. Each array element is an array representing a row.
+     * An empty array is returned if the query results in nothing.
+     * @throws CException execution failed
+     */
+    public function getUnchangedColumns($sid, $sTimestamp, $qTimestamp)
+    {
```

Comment:
@ptelu are you sure this can be rewritten with Yii's query builder?

![image](https://github.com/LimeSurvey/LimeSurvey/assets/2805488/7bef2b54-f4a8-4f5f-a463-bcc2bbca1f3c)

I wonder how we can load column and table data from information_schema without massive performance loss using the query builder.

Reply:
@ptelu can you elaborate on the compatibility problem you are worried of? Can you describe an example when the current solution could go wrong according to you?

Reply:
> @ptelu can you elaborate on the compatibility problem you are worried of? Can you describe an example when the current solution could go wrong according to you?

I learn than without testing you always have surprise when you move a SQL request to MSSQL or PGSQL.

Reply:
It's only for cloud ? Not for CE ?

Reply:
Currently the only use-case will be for Cloud, this is why other RDBMS's are less of a worry. Later on we might work out the compatibliity for other RDBMS systems. At this point we need it for an app and it was implemented as part of CE. We have no UI support for it from core and we are going to use it only from a Cloud feature for the time being.